src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/SimpleStrandSwitchVariantDetector.java

```diff
@@ -93,73 +101,75 @@ private static AlignedContig removeOverlap(final AlignedContig contig) {
+            final CigarElement ce = cigarElements.get(idx);
+            if (ce.getOperator().consumesReadBases()) {
+                if (readBasesConsumed + ce.getLength() < clipLengthOnRead) {
                     readBasesConsumed += ce.getLength();
-                else { // enough read bases would be clipped
-
-                    if ( !ce.getOperator().isAlignment() && !ce.getOperator().equals(CigarOperator.I))
-                        throw new GATKException.ShouldNeverReachHereException("Logic error, should not reach here");
-
-                    // dead with cigar first
-                    final List<CigarElement> resultCEs = threeSections._1();
-                    final int a = readBasesConsumed + ce.getLength() - clipLength;
-                    final CigarOperator op = ce.getOperator().isAlignment() ? CigarOperator.M : CigarOperator.S;
-                    if (clipFrom3PrimeEnd) {
-                        resultCEs.addAll(unclippedCigarElementsForThisAlignment.subList(0, idx));
-                        if (a!=0) {
-                            resultCEs.add( new CigarElement(a, op) );
-                        }
-                        resultCEs.add(new CigarElement(clipLength, CigarOperator.S));
-                    } else {
-                        resultCEs.add(new CigarElement(clipLength, CigarOperator.S));
-                        if (a!=0) {
-                            resultCEs.add( new CigarElement(a, op) );
-                        }
-                        resultCEs.addAll(unclippedCigarElementsForThisAlignment.subList(idx+1, unclippedCigarElementsForThisAlignment.size()));
-                    }
-                    if (!threeSections._3().isEmpty())
-                        resultCEs.addAll(threeSections._3());
-                    newCigar = new Cigar(SvCigarUtils.compactifyNeighboringSoftClippings(resultCEs));
-
-                    // then deal with ref span, note that here we can have only either 'M' or 'I'
-                    refBasesConsumed += ce.getOperator().isAlignment() ? (clipLength - readBasesConsumed)
+                } else { // enough read bases would be clipped, note that here we can have only either 'M' or 'I'
+
+                    if (!ce.getOperator().isAlignment() && !ce.getOperator().equals(CigarOperator.I))
+                        throw new GATKException.ShouldNeverReachHereException("Logic error, should not reach here: operation consumes read bases but is neither alignment nor insertion.\n " +
+                                "Original cigar(" + TextCigarCodec.encode(input.cigarAlong5to3DirectionOfContig) + ")\tclipLengthOnRead(" + clipLengthOnRead + ")\t" + (clipFrom3PrimeEnd ? "clipFrom3PrimeEnd" : "clipFrom5PrimeEnd"));
+
+                    // deal with cigar first
+                    newMiddleSection.add( new CigarElement(clipLengthOnRead, CigarOperator.S) );
+                    final int a = readBasesConsumed + ce.getLength() - clipLengthOnRead;
```

Comment:
Would a more descriptive name for `a` be `basesRemainingOnCigarElementAfterClipping`? Just trying to make sure I understand the algorithm.

Reply:
yes, that's right and updated with documentation